homeassistant/components/peco/sensor.py

```diff
@@ -46,28 +46,23 @@ async def async_setup_entry(
     websession = async_get_clientsession(hass)
     county: str = config_entry.data[CONF_COUNTY]
 
-    async def async_update_data() -> dict[str, float]:
+    async def async_update_data() -> OutageResults:
         """Fetch data from API."""
         try:
-            data = (
-                cast(dict[str, float], await api.get_outage_totals(websession))
+            data: OutageResults = (
+                await api.get_outage_totals(websession)
                 if county == "TOTAL"
-                else cast(
-                    dict[str, float],
-                    await api.get_outage_count(county, websession),
-                )
+                else await api.get_outage_count(county, websession)
             )
         except HttpError as err:
             raise UpdateFailed(f"Error fetching data: {err}") from err
         except BadJSONError as err:
             raise UpdateFailed(f"Error parsing data: {err}") from err
         except asyncio.TimeoutError as err:
             raise UpdateFailed(f"Timeout fetching data: {err}") from err
-        if data["percent_customers_out"] < 5:
-            percent_out = round(
-                data["customers_out"] / data["customers_served"] * 100, 3
-            )
-            data["percent_customers_out"] = percent_out
+        if data.percent_customers_out < 5:
+            percent_out = round(data.customers_out / data.customers_served * 100, 3)
+            data.percent_customers_out = percent_out
```

Comment:
This is not something that belongs in Home Assistant. 
We do not allow for calculated data/features from within Home Assistant.

Reply:
The reason why I'm doing this is because peco does not return any value under 5%, and just shows "less than 5%" on the website. What should I do here?

Reply:
Display the value as provided by the device or service.